### PR TITLE
update script for new server-to-server sharing

### PR DESCRIPTION
**Pull request by schiessle**
set accept to 1 (=true) for all server-to-server shares on update, at this point in time all shares are created by the first version of server-to-server sharing and are accepted if they were added to the table.

### CODE REVIEW

**Comment by karlitschek** [comment]
looks good :+1:

File: apps/files_sharing/lib/migration.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+ /**
+ * ownCloud - migration to new version of the files sharing app
+ *
+ * @copyright (C) 2014 ownCloud, Inc.
```
Comment (nickvergessen) [inline apps/files_sharing/lib/migration.php]: 2015? :see_no_evil:

File: apps/files_sharing/tests/migration.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		parent::__construct();
+
+		$this->connection = \OC::$server->getDatabaseConnection();
+	}
+
+	function testAddAccept() {
+
+		$query = $this->connection->prepare('
+			INSERT INTO `*PREFIX*share_external`
+			(`remote`, `share_token`, `password`, `name`, `owner`, `user`, `mountpoint`, `mountpoint_hash`, `remote_id`, `accepted`)
```
Comment (DeepDiver1975) [inline apps/files_sharing/tests/migration.php]: so basically the same share could be added multiple times? Does that make sense?
I'd expect some unique constraint here then (here as in the schema definition - no in this PR)

File: apps/files_sharing/tests/migration.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+use OCA\Files_Sharing\Tests\TestCase;
+use OCA\Files_Sharing\Migration;
+
+class Test_Files_Sharing_Migration extends TestCase {
```
Comment (DeepDiver1975) [inline apps/files_sharing/tests/migration.php]: please use proper namespace - THX

File: apps/files_sharing/tests/migration.php (hunk 2 above)
Comment (schiesbn) [inline apps/files_sharing/tests/migration.php]: The mountpoint_hash needs to be unique.

File: apps/files_sharing/appinfo/update.php [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+use OCA\Files_Sharing\Migration;
+
+$installedVersion = \OC::$server->getConfig()->getAppValue('files_sharing', 'installed_version');
+
+// Migration OC7 -> OC8
+if (version_compare($installedVersion, '0.6.0', '<')) {
+	$m = new Migration();
```
Comment (PVince81) [inline apps/files_sharing/appinfo/update.php]: inject the DB connection into the Migration object ?

**Comment by PVince81** [comment]
In my test case with master it seemed that `accepted` was already set on 1.
I guess this PR is for cases where for some reason the value wasn't 1 ?

I'm ok with this :+1:

**Comment by PVince81** [comment]
Jenkins PR here: https://github.com/owncloud/core/pull/13339

**Comment by LukasReschke** [comment]
> I guess this PR is for cases where for some reason the value wasn't 1 ?

In 7.0 there was no `accepted`, thus old shares needs to get this applied otherwise they won't appear as accepted. – At least that's what I understand from our code.

**Comment by schiesbn** [comment]
@LukasReschke yes, that's correct

**Comment by schiesbn** [comment]
Test finished successfully

**Comment by jnfrmarks** [comment]
@schiesbn @LukasReschke 

Is there an upgrade test case here that needs to be validated?

**Pull request merged**